tests/unit/samplegen/test_samplegen.py

```diff
@@ -405,27 +393,47 @@ def test_validate_request_multiple_arguments():
 
 def test_validate_request_reserved_request_name():
     with pytest.raises(samplegen.ReservedVariableName):
-        samplegen.Validator().validate_and_transform_request(
-            [{"field": "class.order", "value": "coleoidea"}])
+        samplegen.Validator().validate_and_transform_request(utils.CallingForm.Request,
+                                                             [{"field": "class.order", "value": "coleoidea"}])
 
 
 def test_validate_request_duplicate_input_param():
     with pytest.raises(samplegen.RedefinedVariable):
-        samplegen.Validator().validate_and_transform_request(
-            [{"field": "squid.mantle_mass",
-              "value": "10 kg",
-              "input_parameter": "mantle_mass"},
-             {"field": "clam.mantle_mass",
-              "value": "1 kg",
-              "input_parameter": "mantle_mass"}])
+        samplegen.Validator().validate_and_transform_request(utils.CallingForm.Request,
+                                                             [{"field": "squid.mantle_mass",
+                                                               "value": "10 kg",
+                                                               "input_parameter": "mantle_mass"},
+                                                              {"field": "clam.mantle_mass",
+                                                               "value": "1 kg",
+                                                               "input_parameter": "mantle_mass"}])
 
 
 def test_validate_request_reserved_input_param():
     with pytest.raises(samplegen.ReservedVariableName):
+        samplegen.Validator().validate_and_transform_request(utils.CallingForm.Request,
+                                                             [{"field": "mollusc.class",
+                                                               "value": "cephalopoda",
+                                                               "input_parameter": "class"}])
+
+
+def test_single_request_client_streaming():
```

Comment:
Not quite, it's because methods with client side streaming can't use the positional/keyword attribute notation when describing the request. These methods need a single 'base' dict that's then wrapped in a list and passed to the client method.

In the general case, request construction needs to be much more sophisticated, but for right now it's a quick sanity check.